### ISSUE TITLE
[Question] Capacitor 3 support

### ISSUE THREAD

**Issue by MrPumpking** [calvinckho/capacitor-jitsi-meet#52]
**EDIT**
After some investigation the issue turned out to be related with me using Capacitor v3 while it is not yet supported

**Original issue**
Hi, sorry for asking this question again, I've seen the previous issues but in those people were trying to run Jitsi in browser, while I'm trying to get it working on Android emulator / device.

From my understanding (maybe I'm wrong), if I'm on a native device I should be able to start a Jitsi conference using the native SDK, while on the web I should use Jitsi iFrame / External API (just like you did [here](https://github.com/calvinckho/restvo-app/blob/d6f1f893584bac098d50d0b00c64820e245820ca/src/app/pages/connect/videoconference/videoconference.page.ts#L77), link taken from one of the issues).

If you take a look at the screenshots below, the detected platform is indeed Android, yet I'm getting a warning in Logcat, saying "the web implementation is not available". Is this the expected behaviour or am I missing something?

![image](https://user-images.githubusercontent.com/9134970/116463053-82d61500-a86a-11eb-80ab-2861cc7bc994.png)
![image](https://user-images.githubusercontent.com/9134970/116463234-bf097580-a86a-11eb-82be-6e71ae261689.png)

As for the code it's just a basic setup, as I'm just starting with my app:

```ts
import React, { useState } from 'react';
import { Capacitor, Plugins } from '@capacitor/core';
import 'capacitor-jitsi-meet';

const { Jitsi } = Plugins;

const Home: React.FC = () => {
  const [status1, setStatus1] = useState('');
  const [status2, setStatus2] = useState('');
  const [status3, setStatus3] = useState('');

  const joinCall = async () => {
    const result: any = await Jitsi.joinConference({
      roomName: 'CapacitorTest1',
      url: 'https://meet.jit.si',
      displayName: 'CapacitorUser',
    });
    setStatus1(result.success);

    window.addEventListener('onConferenceJoined', (event) => {
      setStatus2('Conference joined');
    });
    window.addEventListener('onConferenceLeft', (event) => {
      setStatus3('Conference joined');
    });
  };

  return (
    <div>
      <button onClick={() => joinCall()}>Join</button>
      <b>Platform: {Capacitor.getPlatform()}</b>
      <h2>statuses</h2>
      <pre>{JSON.stringify(status1, null, 2)}</pre>
      <pre>{JSON.stringify(status2, null, 2)}</pre>
      <pre>{JSON.stringify(status3, null, 2)}</pre>
    </div>
  );
};

export default Home;


```

**Comment by calvinckho** [comment]
has the android plugin been installed correctly? in Android Studio, do you see capacitor-jitsi-meet listed under the Android project?
![Screen Shot 2021-04-28 at 2 32 00 PM](https://user-images.githubusercontent.com/23274587/116475339-8ea3e600-a82e-11eb-8498-af7a249701ab.png)

**Comment by MrPumpking** [comment]
Yes
![image](https://user-images.githubusercontent.com/9134970/116508643-11c24c00-a8c2-11eb-9bc9-17ed9edc7fe8.png)

**Comment by calvinckho** [comment]
Do you see the folders under my capacitor-jitsi-meet are different than yours? For the sake of comparing, I have /build, /src, and also the file build.gradle under each plugin.

![Screen Shot 2021-04-28 at 11 37 14 PM](https://user-images.githubusercontent.com/23274587/116511019-abfea180-a87a-11eb-98f8-e541a0f13a12.png)

**Comment by MrPumpking** [comment]
Seems so. I'm preparing a repo with reproducible example right now, guess it will be more helpful.

![image](https://user-images.githubusercontent.com/9134970/116511582-eee66680-a8c6-11eb-9326-9ec14431023a.png)

**Comment by MrPumpking** [comment]
While setting up the repo I've decided to check whether Capacitor v3 could be the issue here and turns out it is. Here's the repo:
https://github.com/MrPumpking/nextjs-capacitor-jitsi-meet

There are 3 branches:
- main - contains basic Next.js setup, if you'd like to go through the setup process yourself
- capacitor-v2 - app with capacitor-jitsi-meet and capacitor v2 - works fine
- capacitor-v3 - app with capacitor-jitsi-meet and capacitor v3 - throws "the web implementation is not available" on mobile

To build the app just run `yarn build`, it will build and export static Next.js files to `out` directory. Additionally I had to transpile `capacitor-jitsi-meet` using [`next-transpile-modules`](https://www.npmjs.com/package/next-transpile-modules) as I was getting the following error:

![image](https://user-images.githubusercontent.com/9134970/116541040-3a127080-a8eb-11eb-8dd1-85bbc2f8eb56.png)

If you were to run in to this issue, check `next.config.js` on either capacitor-v2 or capacitor-v3 branch.

**Comment by calvinckho** [comment]
So you were using capacitor version three and that was the problem?￼ once you downgrade your version to v2 you are OK?￼

**Comment by MrPumpking** [comment]
Yes, it's running fine with capacitor v2.

**Comment by MrPumpking** [comment]
Maybe it would be worthwhile to create a separate issue for Capacitor v3 support and update the current docs with information that it is not currently supported? Either way, thank you for your time and for your work on this plugin!

**Comment by calvinckho** [comment]
Sounds good. I will reopen this issue and if you can update the readme doc and create a PR I would appreciate it.

**Comment by dimitri320** [comment]
@calvinckho any updates on this issue? My team and I will take a stab at this issue, if we succeed, we’d be happy to share our code with the community:)

**Comment by calvinckho** [comment]
@dimitri320 if your team can give it a stab, it is greatly appreciated! I do not have any update as I just got back from a trip. If you can get a version working for capacitor 3, I can publish a beta release for the community and if working well a production release.

**Comment by rymeier** [comment]
Whats the status of bringing in support of Capacitor v3. I'm currently building out a platform and I was recently encouraged to upgrade to v3 for other plugins?

**Comment by calvinckho** [comment]
@dimitri320 any luck with upgrading the plugin to work with Capacitor 3?

**Comment by dimitri320** [comment]
@calvinckho yes, we did. My colleague will join this conversation and take it from here.

**Comment by calvinckho** [comment]
thanks @cyril-colin for catching it! I fixed it with 2.0.1 and it should be reflected correctly on npm.
